subsys/net/ip/utils.c

```diff
@@ -932,6 +932,25 @@ bool net_ipaddr_parse(const char *str, size_t str_len, struct sockaddr *addr)
 	return false;
 }
 
+int net_port_set_default(struct sockaddr *addr, uint16_t default_port)
+{
+
+	if (IS_ENABLED(CONFIG_NET_IPV4) && addr->sa_family == AF_INET &&
+	    net_sin(addr)->sin_port == 0) {
+		net_sin(addr)->sin_port = htons(default_port);
+	} else if (IS_ENABLED(CONFIG_NET_IPV6) && addr->sa_family == AF_INET6 &&
+		   net_sin6(addr)->sin6_port == 0) {
+		net_sin6(addr)->sin6_port = htons(default_port);
+	} else if ((IS_ENABLED(CONFIG_NET_IPV4) && addr->sa_family == AF_INET) ||
+		   (IS_ENABLED(CONFIG_NET_IPV6) && addr->sa_family == AF_INET6)) {
+		; /* Port is already set */
+	} else {
+		LOG_ERR("Unknown address family");
+		return -EINVAL;
+	}
+	return 0;
```

Comment:
Please add empty line after closing block, we try to follow that style in zephyr.